cmd/signals.go

```diff
@@ -79,6 +80,7 @@ func handleSignals() {
 			logger.LogIf(context.Background(), err)
 			exit(stopProcess())
 		case osSignal := <-globalOSSignalCh:
+			daemon.SdNotify(false, daemon.SdNotifyStopping)
```

Comment:
We should use 

```
stop := stopProcess()
daemon.SdNotify(false, daemon.SdNotifyReloading)
rerr := restartProcess()
if rerr == nil {
       daemon.SdNotify(false, daemon.SdNotifyReady)
}
```

Should be also called for "serviceRestart" IMO 

And also for `serviceSTop` we must have 

```
daemon.SdNotify(false, daemon.SdNotifyStopping)
```

Reply:
Right, this will allow us to use `Type=notify-reload` in the future for reload synchronization as well (introduced in latest [systemd v253](https://github.com/systemd/systemd/releases/tag/v253)).

And as of missed `case serviceStop:`, my bad, sorry.

Added this and that.

Reply:
Though I think it would be better to send "reloading" notification before calling `stopProcess`?

Reply:
Yes that is correct

Reply:
Let's not spend too much time on the nuances of reload, if we can do it properly within expected semantics then it's fine. 

Otherwise let's not add it there just add Ready when restart is complete?

Reply:
Sure makes sense..